### PR TITLE
Common availability check operation

### PR DESCRIPTION
**Pull request by slemrmartin**
**Issue** https://github.com/RedHatInsights/sources-api/issues/206

This PR moves Source#availability_check from AnsibleTower to providers-common 
(it's refactoring of https://github.com/RedHatInsights/topological_inventory-ansible_tower/pull/95 and https://github.com/RedHatInsights/topological_inventory-ansible_tower/pull/96)

Then it can be used for other repos

---

[TPINVTRY-944](https://projects.engineering.redhat.com/browse/TPINVTRY-944)

### CODE REVIEW

File: lib/topological_inventory/providers/common/operations/source.rb [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+          status, error_message = connection_status
+
+          update_source_and_endpoint(status, error_message)
+
+          logger.availability_check("Completed: Source #{source_id} is #{status}")
+        end
+
+        private
+
+        def required_params
```
Comment (lindgrenj6) [inline lib/topological_inventory/providers/common/operations/source.rb:43]: Should this be a constant?

File: lib/topological_inventory/providers/common/operations/sources_api_client.rb [hunk 2]
```diff
@@ -20,21 +22,23 @@ def init_default_api
...
             endpoint_authentications = api.list_endpoint_authentications(endpoint.id.to_s).data || []
             return if endpoint_authentications.empty?
 
-            auth_id = endpoint_authentications.first.id
+            auth_id = if authtype.nil?
+                        endpoint_authentications.first&.id
+                      else
+                        endpoint_authentications.detect { |a| a.authtype = authtype }&.id
+                      end
+            return if auth_id.nil?
```
Comment (lindgrenj6) [inline lib/topological_inventory/providers/common/operations/sources_api_client.rb:40]: So - I don't see this being used yet. Is this mainly so it'll only use the same auth type when grabbing the auth? Is this mainly to get around the potentially many authentications issue?

Comment (slemrmartin) [reply lib/topological_inventory/providers/common/operations/sources_api_client.rb:40]: @lindgrenj6 authtype is used in Amazon and Azure repos:
- https://github.com/RedHatInsights/topological_inventory-amazon/pull/66/files#diff-03ea31ba8ba5089fdaa66797d8209053L49
- https://github.com/RedHatInsights/topological_inventory-azure/pull/31/files#diff-43c857599ea60dfd8011c249f89703baL49

File: lib/topological_inventory/providers/common/operations/source.rb (hunk 1 above)
Comment (slemrmartin) [reply lib/topological_inventory/providers/common/operations/source.rb:43]: I don't think that overwritable method with list of required params is the best candidate for constant

**Review by lindgrenj6 — APPROVED** [review]

**Review by syncrou — APPROVED** [review]
LGTM 👍

**Pull request merged**